nat.ts

```diff
@@ -28,8 +28,8 @@ module J2ME {
   import Bytecodes = Bytecode.Bytecodes;
   import toHEX = IntegerUtilities.toHEX;
 
-  export function asyncImplOld(returnKind: string, promise: Promise<any>) {
-    return asyncImpl(kindCharacterToKind(returnKind), promise);
+  export function asyncImplOld(returnKind: string, promise: Promise<any>, cleanup?: Function) {
```

Comment:
Hmm, the caller could handle its own cleanup by chaining another function to the promise via *then*. But then it'd have to make sure to return the resolved value. This *cleanup* parameter is more explicit, so I like it better. (But I wish DOM Promise had *finally*, which would be even better, since then the caller could chain it without having to worry about it returning the resolved value.)